src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.cs

```diff
@@ -322,7 +322,24 @@ public bool Join(int millisecondsTimeout)
         /// </summary>
         internal const int OptimalMaxSpinWaitsPerSpinIteration = 64;
 
-        public static void SpinWait(int iterations) => RuntimeImports.RhSpinWait(iterations);
+        public static void SpinWait(int iterations)
+        {
+            if (iterations <= 0)
+                return;
+
+            // Max iterations to be done in RhSpinWait.
+            // RhSpinWait does not switch GC modes and we want to avoid native spinning in coop mode for too long.
+            const int spinWaitCoopThreshold = 10000;
+
+            if (iterations > spinWaitCoopThreshold)
+            {
+                RuntimeImports.RhLongSpinWait(iterations);
```

Comment:
Could you please make a non-inlinable wrapper around RhLongSpinWait so that the PInvoke frame overhead does hit the small spin wait counts?

We do similar thing e.g. for `_Memmove`.

Reply:
Yes. I forgot this part. We do not want this to inline.